x-pack/plugins/grokdebugger/public/sections/grokdebugger/components/custom_patterns_input/custom_patterns_input.js

```diff
@@ -4,47 +4,53 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { uiModules } from 'ui/modules';
-import { InitAfterBindingsWorkaround } from 'ui/compat';
+import React from 'react';
+import {
+  EuiAccordion,
+  EuiCallOut,
+  EuiCodeBlock,
+  EuiFormRow,
+  EuiCodeEditor,
+  EuiSpacer
+} from '@elastic/eui';
 import { EDITOR } from '../../../../../common/constants';
-import { applyEditorOptions } from '../../../../lib/ace';
-import template from './custom_patterns_input.html';
-import './custom_patterns_input.less';
-import 'ui/toggle_panel';
-import 'ace';
 
-const app = uiModules.get('xpack/grokdebugger');
+export function CustomPatternsInput({ value, onChange }) {
+  const sampleCustomPatterns = `POSTFIX_QUEUEID [0-9A-F]{10,11}
+MSG message-id=<%{GREEDYDATA}>`;
 
-app.directive('customPatternsInput', function () {
-  return {
-    restrict: 'E',
-    template: template,
-    scope: {
-      onChange: '='
-    },
-    bindToController: true,
-    controllerAs: 'customPatternsInput',
-    controller: class CustomPatternsInputController extends InitAfterBindingsWorkaround {
-      initAfterBindings($scope) {
-        this.isCollapsed = {
-          action: true
-        };
-        $scope.$watch('customPatternsInput.customPatterns', () => {
-          this.onChange(this.customPatterns);
-        });
-        $scope.aceLoaded = (editor) => {
-          this.editor = editor;
-          applyEditorOptions(editor, EDITOR.PATTERN_MIN_LINES, EDITOR.PATTERN_MAX_LINES);
-        };
-      }
+  return (
+    <EuiAccordion
+      id="customPatternsInput"
+      buttonContent="Custom Patterns"
+      data-test-subj="btnToggleCustomPatternsInput"
+    >
+      <EuiCallOut
```

Comment:
How about adding a `<EuiSpacer size="m" />` above this call out? This will give it a bit of breathing room:

![image](https://user-images.githubusercontent.com/1238659/41756034-13b7b928-758f-11e8-8997-3ef859679612.png)